sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -244,13 +262,35 @@ public Mono<Void> addCredits(int credits) {
         });
     }
 
+    @Override
+    public void addCredit(Supplier<Long> creditSupplier) {
+        assert this.isV2;
+        if (isDisposed()) {
+            throw new RejectedExecutionException(String.format(
+                "connectionId[%s] linkName[%s] Cannot schedule credit flow post the link closure.",
+                handler.getConnectionId(), getLinkName()));
+        }
+        try {
+            dispatcher.invoke(() -> {
+                final long credit = creditSupplier.get();
+                receiver.flow((int) credit);
+                metricsProvider.recordAddCredits((int) credit);
+            });
+        } catch (IOException e) {
+            throw new UncheckedIOException(String.format(
```

Comment:
can we log exception (with the  structured `"connectionId[%s] linkName[%s]` context) here?

Reply:
not logging here was intentional to avoid storm of duplicate logging. In V2 we catch this in the call-site and log [here](https://github.com/anuchandy/azure-sdk-for-java/blob/messaging-integrate-new-core-types-1/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/CreditAccountingStrategy.java#L62-L66) to cut the noise.

Reply:
By the way, thank you for the comment; this reminds us to log better in the call site (i.e., to include connection-receiver identifiers)